reactive-streams/src/main/scala/fs2/interop/reactivestreams/StreamSubscriber.scala

```diff
@@ -72,23 +74,23 @@ final class StreamSubscriber[F[_]: ConcurrentEffect, A](val sub: StreamSubscribe
 }
 
 object StreamSubscriber {
-  def apply[F[_]: ConcurrentEffect, A]: F[StreamSubscriber[F, A]] =
+  def apply[F[_]: Async, A]: F[StreamSubscriber[F, A]] =
```

Comment:
This will break binary compatibility.  Leaving a private overload with `ConcurrentEffect` will still complain about a missing static method for Java interop, I think.  I think one of three choices is necessary:

1. Create a new method for the `Async`, which the old one could call.
2. Break Java interop, which is probably fine, but reactivestreams is a place I'd most expect people to do weird Java shit.
3. Keep the constraint that's now stronger than needed.

Reply:
I like the first one. And maybe add `@deprecated` to the old method?

Reply:
On the other hand there's also `fs2.interop.reactivestreams` package object with a couple of methods having `ConcurrentEffect` constraint. So either 2 or 3.

Reply:
Decided to keep the old constraint.